hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/records/RecordsSuite.java

```diff
@@ -299,6 +311,55 @@ final Stream<DynamicTest> blck001And002And003And004ReturnsCorrectBlockProperties
                 }));
     }
 
+    @DisplayName("Block Hash Returns The Hash Of The Latest 256 Blocks")
+    @RepeatableHapiTest(NEEDS_VIRTUAL_TIME_FOR_FAST_EXECUTION)
+    final Stream<DynamicTest> blockHashReturnsTheHashOfTheLatest256Blocks() {
+        final var contract = "EmitBlockTimestamp";
+        return hapiTest(
+                newKeyNamed(SECP_256K1_SOURCE_KEY).shape(SECP_256K1_SHAPE),
+                uploadInitCode(contract),
+                contractCreate(contract).gas(4_000_000L),
+                cryptoCreate(ACCOUNT).balance(6 * ONE_MILLION_HBARS),
+                cryptoCreate(RECEIVER),
+                cryptoTransfer(tinyBarsFromAccountToAlias(GENESIS, SECP_256K1_SOURCE_KEY, ONE_HUNDRED_HBARS)),
+                withOpContext((spec, opLog) -> {
+                    doNTransfers(spec, 256);
+                    waitUntilStartOfNextAdhocPeriod(2_000L);
+                    final var ethCall = ethereumCall(contract, "getAllBlockHashes")
+                            .logged()
+                            .gasLimit(4_000_000L)
+                            .via("blockHashes");
+                    final var blockHashRes = getTxnRecord("blockHashes").logged();
+                    allRunFor(spec, ethCall, waitUntilStartOfNextAdhocPeriod(2_000L), blockHashRes);
+                    assertTrue(blockHashRes
+                            .getResponseRecord()
+                            .getContractCallResult()
+                            .getErrorMessage()
+                            .isEmpty());
+                    final var res = blockHashRes
+                            .getResponseRecord()
+                            .getContractCallResult()
+                            .getContractCallResult()
+                            .substring(64);
```

Comment:
What's in the first 8 bytes you're lopping off here? (I think that's what's going on.)

Reply:
I trim the headers so I can have only the result bytes

Reply:
oh! what's in the headers? (I didn't know there _were_ headers ...)